docs/shared/react-tutorial/4-workspace-optimization.md

```diff
@@ -0,0 +1,231 @@
+
+{% callout type="note" title="More Cache Details" %}
+See the [Core Features documentation page for Cache Task Results](/core-features/cache-task-results) for more information on caching.
+{% /callout %}
+
+## The Task Graph
+
+Next, run the command `npx nx build store`:
+
+```bash
+> npx nx build store
+
+   ✔    1/1 dependent project tasks succeeded [0 read from cache]
+
+   Hint: you can run the command with --verbose to see the full dependent project outputs
+
+ ———————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————
+
+
+> nx run store:build:production
+
+Entrypoint main 139 KiB = runtime.54e36ebee261465d.js 1.19 KiB main.623d91691bdb6754.css 42 bytes main.94f9a4a3cec4f056.js 138 KiB
+Entrypoint polyfills 93.5 KiB = runtime.54e36ebee261465d.js 1.19 KiB polyfills.bd0d0abec287a28e.js 92.3 KiB
+Entrypoint styles 1.19 KiB = runtime.54e36ebee261465d.js 1.19 KiB styles.ef46db3751d8e999.css 0 bytes
+chunk (runtime: runtime) main.623d91691bdb6754.css, main.94f9a4a3cec4f056.js (main) 145 KiB (javascript) 50 bytes (css/mini-extract) [initial] [rendered]
+chunk (runtime: runtime) polyfills.bd0d0abec287a28e.js (polyfills) 301 KiB [initial] [rendered]
+chunk (runtime: runtime) styles.ef46db3751d8e999.css (styles) 50 bytes (javascript) 80 bytes (css/mini-extract) [initial] [rendered]
+chunk (runtime: runtime) runtime.54e36ebee261465d.js (runtime) 3.23 KiB [entry] [rendered]
+webpack compiled successfully (06e95dfdacea84c7)
+
+ ———————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————
+
+ >  NX   Successfully ran target build for project store and 1 task(s) it depends on (5s)
+```
+
+Notice the line here:
+
+```bash
+   ✔    1/1 dependent project tasks succeeded [0 read from cache]
+```
+
+This is because your `store` project is dependent on your `products` project, which is also buildable. By default Nx is configured to run (or pull results from cache) for any dependent project's `build`s.
+
+{% callout type="note" title="Customizing Task Inputs" %}
+When you generated your workspace, those scripts set reasonable `targetDefaults` in your `nx.json` file.
+
+See our [guide for Customizing Inputs and Named Inputs](/more-concepts/customizing-inputs) for more information on how to customize these to better fit your workspace.
+{% /callout %}
+
+This can be particularly helpful in relying on your Nx graph to dynamically maintain task dependencies, rather than having to manually maintain those task dependencies as your workspace continues to grow.
```

Comment:
```suggestion
This feature allows the Nx graph to dynamically maintain task dependencies, rather than having to manually maintain those task dependencies as your workspace continues to grow.
```